### ISSUE TITLE
workaround unity 2020

### ISSUE THREAD

**Issue by agrobost** [OneSignal/OneSignal-Unity-SDK#332]
As many may have noticed, the OneSignal-Unity-SDK doesn't work on unity 2020..
I found how to get around the problem, I share my solution with you, it's not great, but better than nothing while waiting for an update.

1. Comment the method createOneSignalAndroidManifest() in OneSignalEditorScript_Android.cs

2. Remove the file Plugins/Android/OneSignalConfig/AndroidManifestTemplate.xml
3. Keep the manifest generated (Plugins/Android/OneSignalConfig/AndroidManifest.xml) and remove the tag
```xml
<uses-sdk android:minSdkVersion="15" />
```
4. Rename the folder Plugins/Android/OneSignalConfig to Plugins/Android/OneSignalConfig.androidlib

_Note_
https://docs.unity3d.com/2020.1/Documentation/Manual/AndroidAARPlugins.html
We can no longer use "use-sdk" in the androidManifest
https://stackoverflow.com/questions/19997509/android-studio-why-are-minsdkversion-and-targetsdkversion-specified-both-in-and

**Comment by jkasten2** [comment]
@agrobost Thanks for digging in and sharing. We just released [OneSignal-Unity-SDK 2.13.3](https://github.com/OneSignal/OneSignal-Unity-SDK/releases/tag/2.13.3) with a fix for this.

> 4. Rename the folder Plugins/Android/OneSignalConfig to Plugins/Android/OneSignalConfig.androidlib

We a similar rename, instead we went with `OneSignalConfig.plugin` but the effect is the same.

> 1. Comment the method createOneSignalAndroidManifest() in OneSignalEditorScript_Android.cs

We accounted for this by using a GUID lookup (so the folder can be safely moved), with a fallback to search by the new  `OneSignalConfig.plugin` name.

> 3. Keep the manifest generated (Plugins/Android/OneSignalConfig/AndroidManifest.xml) and remove the tag
`<uses-sdk android:minSdkVersion="15" />`
...
*Note*
https://docs.unity3d.com/2020.1/Documentation/Manual/AndroidAARPlugins.html
We can no longer use "use-sdk" in the androidManifest
https://stackoverflow.com/questions/19997509/android-studio-why-are-minsdkversion-and-targetsdkversion-specified-both-in-and

This is something that needs to stay as there isn't a `build.gradle` in the repo to define the `minSdkVersion` via Gradle. [The answer in stackoverflow](https://stackoverflow.com/a/20002844/1244574) notes the following:
> Gradle overrides the manifest values, and I prefer to update the build.gradle file rather than the manifest.

So if Unity did make a change to auto generate a `build.gradle` at build time with a `minSdkVersion` value this is ok since it will safely just ignore the `AndroidManifest.xml` one. The important this is that `minSdkVersion` is provided some where, otherwise it will be defaulted to 1 which will create build issues.

How the Android native library and `AndroidManifest.xml` are added the the project is something we are looking into modernizing. We may be able to get Unity or Gradle to pull down and use the public `OneSignal-Android-SDK` so we don't even require the `OneSignalConfig.plugin` folder in the future.
